src/subscriber.rs

```diff
@@ -125,6 +125,44 @@ impl Drop for Subscriber {
 }
 
 impl Subscriber {
+    /// Creates a future that resolves to the next value of the
+    /// subscriber, or None if the backing `Db` shuts down
+    pub fn next(&mut self) -> impl Future<Output = Option<Event>> + '_ {
+        Next { subscriber: self }
+    }
+
+    fn poll_next(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Option<Event>> {
```

Comment:
This API matches the current definition of the `Stream` trait in Futures 0.3, so if that ends up stabilizing as-is then a Stream implementation will be "free"

Otherwise some changes will need to be made later down the line, but it's a private method anyway